src/entities/entity.ts

```diff
@@ -0,0 +1,73 @@
+import { Component } from '../components/component';
+
+/*
+ * Custom error class for duplicate component errors.
+ */
+class DuplicateComponentError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = 'DuplicateComponentError';
+  }
+}
+
+/**
+ * Represents an entity in the system.
+ */
+export class Entity {
+  /**
+   * The unique identifier for the entity.
+   */
+  id: string;
+
+  /**
+   * A map of components associated with the entity.
+   */
+  components: Map<string, Component> = new Map<string, Component>();
+
+  /**
+   * Creates a new entity with the given identifier.
+   * @param {string} id - The unique identifier for the entity.
+   */
+  constructor(id: string) {
+    this.id = id;
+  }
+
+  /**
+   * Adds a component to the entity.
+   * @param {Component} component - The component to be added.
+   * @throws {DuplicateComponentError} If a component of the same type already exists.
+   */
+  addComponent(component: Component): void {
```

Comment:
Have this function return itself, so you can chain calls like:

```
new Entity()
  .addComponent(...)
  .addComponent(...)
  ...
```

Reply:
Done 02c82bf85d9fc18543ef25910ef78ece74c22e55